woocommerce/woocommerce-gutenberg-products-block
# Update @wordpress/env to 4.0.2 and test default config with core property set to null.

A while ago there was an official WordPress docker container update that resulted in some downstream permission problems with `@wordpress/env` (see #3972). The permission issues are fixed in the latest patch version of `wp-env` so this PR updates the following:

- Restores the default `.wp-env.json` config to load the latest WordPress version release available (via the `core` property set to `null`). One advantage of doing this is the test runs for latest WP version should be a bit faster (but in practice I'm not seeing any change).
- While testing some other issues I also noticed that if I installed the latest `@wordpress/api-fetch` package directly I could get the latest types available there which improved some type script typing of our apiFetch implementation because the definitely typed install didn't have latest types.

## To Test

The changes here doesn't impact user facing code. So I did the following:

- Verified e2e tests pass locally for the default configuration ✅ 
- Wait for e2e tests to all be green via GH. 

There's still an issue with some flakiness with the GitHub e2e workflow testing with WP 5.7 and the Gutenberg plugin active. The failing test is not reproducible in manual testing so if after a couple retries it still doesn't pass I'm going to consider it okay and merge anyways. I took a brief gander and there's something with the `Company` text label xPath search not working. It might be good to see if tightening the selector there might help but it can be done in a separate PR.
